Build failure on Electron 31+

when building on Electron 31, we get:

```
npm ERR! ../../@axosoft/nan/nan.h:208:12: error: no template named 'CopyablePersistentTraits' in namespace 'v8'; did you mean simply 'CopyablePersistentTraits'?
npm ERR!     public v8::CopyablePersistentTraits<T> {};
npm ERR!            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm ERR!            CopyablePersistentTraits
npm ERR! ../../@axosoft/nan/nan.h:207:7: note: 'CopyablePersistentTraits' declared here
npm ERR! class CopyablePersistentTraits :
npm ERR!       ^
npm ERR! ../../@axosoft/nan/nan.h:208:12: error: circular inheritance between 'v8::CopyablePersistentTraits<T>' and 'CopyablePersistentTraits<T>'
npm ERR!     public v8::CopyablePersistentTraits<T> {};
```

This is probably related to nan that needs to include the latest fix of version 2.20

needs the PR https://github.com/nodejs/nan/pull/970

as mention also in: https://github.com/nodegit/nodegit/pull/1998#issuecomment-2185009918

@ianhattendorf

--- a/package.json
+++ b/package.json
@@ -38,7 +38,7 @@
     "node": ">= 16"
   },
   "dependencies": {
-    "@axosoft/nan": "^2.19.0-gk.1",
+    "@axosoft/nan": "^2.20.0-gk.1",
     "@mapbox/node-pre-gyp": "^1.0.8",
     "fs-extra": "^7.0.0",
     "got": "^11.8.6",

